### PR TITLE
Add endpoint to list type action

### PR DESCRIPTION
**Pull request by lucaslinhares**
Add a new endpoint to get a type action.

### CODE REVIEW

File: temba/externals/views.py [hunk 1]
```diff
@@ -77,3 +79,23 @@ class Delete(DependencyDeleteModal):
...
+
+
+class GetExternalTypes(APIView):
+    authentication_classes = []
+    permission_classes = []
+    pagination_class = None
+    renderer_classes = [JSONRenderer]
+    throttle_classes = []
+
+    @action(methods=["get"], detail=False)
```
Comment (Sandro-Meireles) [inline temba/externals/views.py:91]: Is this action really necessary? Here's an example of how an APIview works: https://www.django-rest-framework.org/api-guide/views/

**Review by Sandro-Meireles — CHANGES_REQUESTED** [review]

**Review by Sandro-Meireles — APPROVED** [review]

File: temba/externals/models.py [hunk 2]
```diff
@@ -1,5 +1,9 @@
 from abc import ABCMeta, abstractproperty
 
+from pathlib import Path
```
Comment (elitonzky) [inline temba/externals/models.py:3]: import Path is not being used, to make the code cleaner, unused imports are removed.

File: temba/externals/views.py [hunk 3]
```diff
@@ -4,11 +4,15 @@
 from django.utils.html import mark_safe
 from django.utils.translation import ugettext_lazy as _
 from django.urls import reverse
+from rest_framework.decorators import action
```
Comment (elitonzky) [inline temba/externals/views.py:7]: `import action` is not being used, to make the code cleaner, unused imports are removed.

File: temba/externals/views.py [hunk 4]
```diff
@@ -77,3 +81,23 @@ class Delete(DependencyDeleteModal):
...
+    permission_classes = []
+    pagination_class = None
+    renderer_classes = [JSONRenderer]
+    throttle_classes = []
+
+    def get(self, request, slug=None):
+        from temba.externals.types import TYPES
+
+        external_type = TYPES.get(slug)
+
```
Comment (elitonzky) [inline temba/externals/views.py:97]: line 97 contains whitespace at line.

File: temba/externals/views.py [hunk 5]
```diff
@@ -77,3 +81,23 @@ class Delete(DependencyDeleteModal):
...
+    renderer_classes = [JSONRenderer]
+    throttle_classes = []
+
+    def get(self, request, slug=None):
+        from temba.externals.types import TYPES
+
+        external_type = TYPES.get(slug)
+        
+        if external_type is None:
+            return Response (status=status.HTTP_404_NOT_FOUND)
```
Comment (elitonzky) [inline temba/externals/views.py:99]: on line 99, there is white space before '('
`return Response (status=status.HTTP_404_NOT_FOUND)` 
try:
`return Response(status=status.HTTP_404_NOT_FOUND)`

File: temba/externals/models.py (hunk 2 above)
Comment (lucaslinhares) [reply temba/externals/models.py]: Fixed

File: temba/externals/models.py [hunk 6]
```diff
@@ -1,5 +1,9 @@
...
+import json
+import os, sys
```
[the remark this answers is not in the document]
Comment (lucaslinhares) [reply temba/externals/models.py]: Fixed

File: temba/externals/views.py (hunk 3 above)
Comment (lucaslinhares) [reply temba/externals/views.py]: Fixed

File: temba/externals/views.py (hunk 4 above)
Comment (lucaslinhares) [reply temba/externals/views.py]: Fixed

File: temba/externals/views.py [hunk 7]
```diff
@@ -77,3 +81,23 @@ class Delete(DependencyDeleteModal):
...
+
+        actions = external_type.get_actions()
+        return Response(actions, status=status.HTTP_200_OK)
+
```
[the remark this answers is not in the document]
Comment (lucaslinhares) [reply temba/externals/views.py]: Fixed

**Review by elitonzky — APPROVED** [review]

**Pull request merged**